control_msgs/msg/JointJog.msg

```diff
@@ -0,0 +1,18 @@
+# Used in time-stamping the message.
+Header header
+ 
+# Name list of the joints. You don't need to specify all joints of the
+# robot. Joint names are case-sensitive.
+string[] joint_names
+ 
+# Relative displacement of the joints listed in joint_names.
+# The order must be identical.
+# If positions, units are rad for revolute joints, meters for linear joints.
+# If velocities, units are rad/s or m/s.
+float64[] deltas
+
+# Time information may be required.
```

Comment:
IMHO it must be populated.
Otherwise there is no notion of how hast the jogging motion should be,

Reply:
Well, in my implementation, the spin rate of the higher-level controller is specified in a yaml file, so this time field isn't relevant to me.

Reply:
In all jogging interfaces I know, there is a way to select the (profiled) velocity.
Otherwise you cannot move fast for corse movements and slow for precise motions.

Reply:
Isn't the time step almost always constant for a controller, and we can assume here (through documentation) that the user is sending these commands at the required time step? If messages are dropped or a node goes down, its up to the higher level logic to handle errors. I don't think we need to repeatedly send the control frequency and error threshold in every message, that's a waste of data.

Reply:
Is the position_delta supposed to be valid for one step in the control cycle?
I don't think so.

In Care-O-bot the hardware is running at a 10kHz and 20 kHz rate. The high-level controller is running at 50Hz to 100Hz. So for the user it not easy to tell the actual rate.

In many cases jogging is controlled by the UI, where the freuqency might be much slower..